### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by jwpeterson**
Inspired by adfd541b9, I found [scspell](https://github.com/myint/scspell) and used it to fix a bunch of typos in our documentation.

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -1955,14 +1955,13 @@ Optional Features:
   --enable-legacy-using-namespace
                           add "using namespace libMesh" to libMesh headers
   --enable-everything     enable all non-conflicting options
-  --enable-unique-id      build with unique id suppport
+  --enable-unique-id      build with unique id support
```
Comment (roystgnr) [inline configure]: Ha!  We've been copying and pasting that triple p around for a decade, haven't we?

File: configure [hunk 2]
```diff
@@ -2028,7 +2027,7 @@ Optional Features:
   --enable-fparser-debugging
                           Build fparser with bytecode debugging functions
   --disable-cppunit       Build without cppunit C++ unit testing support
-  --disable-nanoflann     build without nanoflann KD-tree suppport
+  --disable-nanoflann     build without nanoflann KD-tree support
```
Comment (roystgnr) [inline configure]: So how many false positives did you end up with from scspell?  I can't imagine it has "nanoflann", "fparser", and "cppunit" in its dictionary.

File: include/enums/enum_eigen_solver_type.h [hunk 3]
```diff
@@ -71,8 +71,8 @@ enum PositionOfSpectrum {LARGEST_MAGNITUDE=0,
                          LARGEST_IMAGINARY,
                          SMALLEST_IMAGINARY,
                          TARGET_IMAGINARY,
-
-                         INVALID_Postion_of_Spectrum};
+                         INVALID_Postion_of_Spectrum,
+                         INVALID_POSITION_OF_SPECTRUM};
```
Comment (roystgnr) [inline include/enums/enum_eigen_solver_type.h]: Ok, this worries me.  Are there corresponding changes in the code to accept the new enum option and to spit out a deprecated warning if we see the old option?  Ctrl-F isn't finding them for me, even after I expand all the foo_eigen_bar.C diffs.

File: src/mesh/mesh_communication_global_indices.C [hunk 4]
```diff
@@ -309,7 +309,7 @@ void MeshCommunication::assign_global_indices (MeshBase & mesh) const
 
     communicator.allgather (my_max, /* identical_buffer_sizes = */ true);
 
-    // Be cereful here.  The *_upper_bounds will be used to find the processor
```
Comment (roystgnr) [inline src/mesh/mesh_communication_global_indices.C]: The irony is delicious.

**Comment by roystgnr** [comment]
:+1: after my questions about the enum are resolved.  This is great!

Did you script up any kind of ```git blame | cut | sort | uniq -c``` on the diff, by any chance?  I recognize an embarrassing number of those typos as my own, but I'm not sure if that's availability bias or if I'd really win the Most Illiterate LibMesh Developer booby prize.  I'm not going to check myself, though, because the only way the results could come out are "mean" or "sad"...

File: include/enums/enum_eigen_solver_type.h (hunk 3 above)
Comment (jwpeterson) [reply include/enums/enum_eigen_solver_type.h]: The old enum was never used anywhere, so I didn't really know what else to do to support it?

File: configure (hunk 2 above)
Comment (jwpeterson) [reply configure]: There were definitely *lots* of false positives. But scspell makes it pretty painless to update ~/.scspell/dictionary.txt, which is a human-readable text file with all your local words. So for common words like "libmesh" etc, you only have to add them to the dictionary once. Most of the false positives were triggered by our variable names in code, and those are much harder to ignore.

**Comment by jwpeterson** [comment]
> Did you script up any kind of git blame | cut | sort | uniq -c on the diff, by any chance?

I did not. `git blame` isn't very useful in this regard, as it only tells you the *last* person to touch a line... `git log --reverse -Ssuppport` shows that this typo dates back to d7a66045d, i.e. the "Initial import of sources" commit from 2003.

File: include/enums/enum_eigen_solver_type.h (hunk 3 above)
Comment (roystgnr) [reply include/enums/enum_eigen_solver_type.h]: Oh, was this just one of those habits we get into where whenever we make an enum we give it an "INVALID_WHATEVER" option too?  In that case we're good to go.

**Pull request merged**